sympy/solvers/solveset.py

```diff
@@ -521,6 +521,15 @@ def _solve_as_rational(f, symbol, domain):
         invalid_solns = _solveset(h, symbol, domain)
         return valid_solns - invalid_solns
 
+def _solve_hyperbolic(f, symbol, domain):
+
+    # transform the hyperbolic functions into non hyperbolic
+    # redo the calulcations from the beginning
```

Comment:
There should be a docstring like for `_solve_rational` above.

The word `calulcations` is misspelt. Perhaps the comment should just say `rewrite hyperbolic functions as exp`

You can see where the equivalent is done in `solve` here:
https://github.com/sympy/sympy/blob/f8581068eb233026b8871ba88b37f1eab59aafc4/sympy/solvers/solvers.py#L1030-L1032

Note that it only rewrites hyperbolic functions and not anything else as exp. Probably it's better to use `rewrite` rather than `replace` but to specify which functions you want to rewrite:
```julia
In [6]: from sympy.functions.elementary.hyperbolic import HyperbolicFunction                                                      

In [7]: (cosh(x) + sinh(x) + y**z).rewrite(HyperbolicFunction, exp)                                                               
Out[7]: 
 z    x
y  + ℯ 
```
Otherwise you have:
```julia
In [8]: (cosh(x) + sinh(x) + y**z).rewrite(exp)                                                                                   
Out[8]: 
 x    z⋅log(y)
ℯ  + ℯ 
```